samples/basic/shell/src/AppRouter.tsx

```diff
@@ -9,13 +11,21 @@ function Loading() {
 }
 
 export function AppRouter() {
+    const { toastState, addToast } = useToastContainer();
+
+    const handleShowToast = useCallback((message: unknown) => {
+        addToast(message as string);
+    }, [addToast]);
+
+    useApplicationEventBusListener("show-toast", handleShowToast);
```

Comment:
It seems like we're losing a lot by going with the useApplicationEventBusListener approach. 

Instead of having strong type checking, making sure props expected in the shell and props send by the remotes are the same, we call a hook that accepts an unknown payload, and receive an unknown payload in the shell. 

It looks like the redux implementation we had before Typescript.

Reply:
In this simple message where the only props is message, it doesn't matter much. But at one point we might have more customization, such as toast type, toasts we don't want to dismiss, toasts with links or actions inside of it

Reply:
at the very least i would create another small hook that does all this: 
```
import { useCallback } from "react";
import { useApplicationEventBusDispatcher } from "./eventBus.ts";

export interface ToastState {
    message:string
}

export type ToastCreationPayload = ToastState | string;

export function useToast() {
    const dispatch = useApplicationEventBusDispatcher();

    return useCallback((message: ToastCreationPayload) => {
        dispatch("show-toast", message);
    }, [dispatch]);
}

export function useToastListener(listener: (payload: ToastCreationPayload) => void) { 
    useApplicationEventBusListener("show-toast", listener as  (payload: ToastCreationPayload) => void);
} 
```

Reply:
Done!